src/main/kotlin/com/exactpro/th2/codec/hand/processor/ProtobufHandProcessor.kt

```diff
@@ -0,0 +1,107 @@
+ * limitations under the License.
+ */
+package com.exactpro.th2.codec.hand.processor
+
+import com.exactpro.th2.codec.hand.CodecHandSettings
+import com.exactpro.th2.codec.hand.processor.ProtobufHandProcessor.Companion.Context
+import com.exactpro.th2.codec.util.ERROR_CONTENT_FIELD
+import com.exactpro.th2.codec.util.ERROR_TYPE_MESSAGE
+import com.exactpro.th2.common.grpc.AnyMessage
+import com.exactpro.th2.common.grpc.EventID
+import com.exactpro.th2.common.grpc.Message
+import com.exactpro.th2.common.grpc.MessageID
+import com.exactpro.th2.common.grpc.MessageMetadata
+import com.exactpro.th2.common.grpc.RawMessage
+import com.exactpro.th2.common.grpc.RawMessageMetadata
+import com.exactpro.th2.common.grpc.Value
+import com.exactpro.th2.common.value.toValue
+import com.google.protobuf.ByteString
+
+class ProtobufHandProcessor(
+    settings: CodecHandSettings
+): HandProcessor<RawMessage, Context, AnyMessage>(
+    settings
+) {
+    override val RawMessage.bodyAsString: String
+        get() = body.toStringUtf8()
+
+    override fun RawMessage.createContext(): Context = Context(this)
+
+    override fun createRaw(context: Context, subSequence: Int, rawData: String): AnyMessage =
+        AnyMessage.newBuilder().apply {
+            val rawMsgMetaData = context.rawMetaDataBuilder
+                .setId(
+                    context.messageIdBuilder.clearSubsequence()
+                        .addSubsequence(subSequence)
+                )
+                .build()
+            val builderForValue = RawMessage.newBuilder().apply {
+                setMetadata(rawMsgMetaData)
+                context.parentEventId?.let(::setParentEventId)
+                setBody(ByteString.copyFrom(rawData.toByteArray()))
+            }
+            setRawMessage(builderForValue)
+        }.build()
+
+    override fun createParsed(context: Context, subSequence: Int, key: String, value: Any): AnyMessage =
+        createParsed(context, DEFAULT_MESSAGE_TYPE, subSequence, key, value)
+
+    override fun createError(context: Context, subSequence: Int, text: String): AnyMessage =
+        createParsed(context, ERROR_TYPE_MESSAGE, subSequence, ERROR_CONTENT_FIELD, text)
+
+    private fun createParsed(context: Context, type: String, subSequence: Int, key: String, value: Any): AnyMessage =
+        AnyMessage.newBuilder().apply {
+            val msgMetaData = context.metaDataBuilder.setId(
+                context.messageIdBuilder.clearSubsequence().addSubsequence(subSequence)
+            ).setMessageType(type).build()
+
+            setMessage(
+                Message.newBuilder().apply {
+                    setMetadata(msgMetaData)
+                    context.parentEventId?.let(::setParentEventId)
+                    putFields(key, convertToValue(value))
+                }
+            )
+        }.build()
+
+    private fun convertToValue(value: Any): Value = value.toValue()
+
+    companion object {
+        class Context(message: RawMessage) {
```

Comment:
You can define the class inside the `ProtobufHandProcessor`. Right now it is in `ProtobufHandProcessor.Companion` class